server/http_test.go

```diff
@@ -842,3 +949,593 @@ func TestGetSubscriptionsForChannel(t *testing.T) {
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID: "aaaaaaaaaaaaaaaaaaaaaaaaab",
+							Filters: SubscriptionFilters{
+								Events:     NewStringSet("jira:issue_created"),
+								Projects:   NewStringSet("myproject"),
+								IssueTypes: NewStringSet("10001"),
+							},
+						},
+					}), t),
+		},
+		"Editing subscription template, no issue types provided": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "somename", "id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaac", "filters": {"events": ["jira:issue_created"], "projects": ["otherproject"], "issue_types": []}}`,
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID: "aaaaaaaaaaaaaaaaaaaaaaaaab",
+							Filters: SubscriptionFilters{
+								Events:     NewStringSet("jira:issue_created"),
+								Projects:   NewStringSet("myproject"),
+								IssueTypes: NewStringSet("10001"),
+							},
+						},
+					}), t),
+		},
+		"Editing subscription template, GetProject mocked error. Existing sub has nonexistent project.": {
+			subscriptionTemplate:       fmt.Sprintf(`{"instance_id": "jiraurl1", "id": "subaaaaaaaaaabbbbbbbbbbccc", "name": "subscription name", "channel_id": "channelaaaaaaaaaabbbbbbbbb", "filters": {"events": ["jira:issue_created"], "projects": ["%s"], "issue_types": ["10001"]}}`, nonExistantProjectKey),
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID: "subaaaaaaaaaabbbbbbbbbbccc",
+							Filters: SubscriptionFilters{
+								Events:     NewStringSet("jira:issue_updated"),
+								Projects:   NewStringSet("myproject"),
+								IssueTypes: NewStringSet("10001"),
+							},
+						},
+					}), t),
+		},
+		"Editing subscription template, GetProject mocked error. Existing sub has existing project.": {
+			subscriptionTemplate:       fmt.Sprintf(`{"instance_id": "jiraurl1", "id": "subaaaaaaaaaabbbbbbbbbbccc", "name": "subscription name", "channel_id": "channelaaaaaaaaaabbbbbbbbb", "filters": {"events": ["jira:issue_created"], "projects": ["%s"], "issue_types": ["10001"]}}`, nonExistantProjectKey),
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID: "subaaaaaaaaaabbbbbbbbbbccc",
+							Filters: SubscriptionFilters{
+								Events:     NewStringSet("jira:issue_updated"),
+								Projects:   NewStringSet("myproject"),
+								IssueTypes: NewStringSet("10001"),
+							},
```

Comment:
we can create a util function to get `Filters` value with different event types as this is used at many places with the same values. check everywhere